src/Model/VerifyEmailSignatureComponents.php

```diff
@@ -46,4 +63,82 @@ public function getExpiresAt(): \DateTimeInterface
     {
         return $this->expiresAt;
     }
+
+    /**
+     * Get the translation message for when a signature expires.
+     *
+     * This is used in conjunction with the getExpirationMessageData() method.
+     * Example usage in a Twig template:
+     *
+     * <p>{{ components.expirationMessageKey|trans(components.expirationMessageData) }}</p>
+     *
+     * symfony/translation is required to translate into a non-English locale.
+     *
+     * @throws GeneratedAtNotSetException
+     */
+    public function getExpirationMessageKey(): string
+    {
+        $interval = $this->getExpiresAtIntervalInstance();
+
+        switch ($interval) {
+            case $interval->y > 0:
+                $this->transInterval = $interval->y;
+
+                return '%count% year|%count% years';
+            case $interval->m > 0:
+                $this->transInterval = $interval->m;
+
+                return '%count% month|%count% months';
+            case $interval->d > 0:
+                $this->transInterval = $interval->d;
+
+                return '%count% day|%count% days';
+            case $interval->h > 0:
+                $this->transInterval = $interval->h;
+
+                return '%count% hour|%count% hours';
+            default:
+                $this->transInterval = $interval->i;
+
+                return '%count% minute|%count% minutes';
+        }
+    }
+
+    /**
+     * @throws GeneratedAtNotSetException
+     */
+    public function getExpirationMessageData(): array
+    {
+        $this->getExpirationMessageKey();
+
+        return ['%count%' => $this->transInterval];
+    }
+
+    /**
+     * Get the interval that the signature is valid for.
+     *
+     * @throws GeneratedAtNotSetException
+     *
+     * @psalm-suppress PossiblyFalseArgument
+     */
+    public function getExpiresAtIntervalInstance(): \DateInterval
+    {
+        if (null === $this->generatedAt) {
+            throw new GeneratedAtNotSetException();
```

Comment:
I wasn't totally clear the first time. I actually think that we do not need this custom class. I can't think of any reason why a developer would legitimately need to catch this. Probably all we need as a `\LogicException()`

Reply:
done